winml/lib/Api.Ort/OnnxruntimeErrors.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#include "pch.h"
+#include "core/providers/winml/winml_provider_factory.h"
+
+#ifdef _WIN32
+inline HRESULT OrtErrorCodeToHRESULT(OrtErrorCode status) noexcept {
```

Comment:
* Should we inline error handling, which is not in the main execution path anyways?
* Must be of internal linkage (put into an anonymous namespace. `static` would also work), else linking will fail if the file is included in more than one place

Reply:
inline methods can have multiple definitions and AFAIK are generally implemented as a compiler detail. They have static linkage, and should not break the build if included in a header.

---
In reply to: [368192201](https://github.com/microsoft/onnxruntime/pull/2869#discussion_r368192201) [](ancestors = 368192201)

Reply:
Yes, we discussed it here :) I was mistaken:

```
An inline function or variable (since C++17) with external linkage (e.g. not declared static)
has the following additional properties:
* There may be more than one definition of an inline function or variable (since C++17) in
the program as long as each definition appears in a different translation unit and (for non-
static inline functions and variables (since C++17)) all definitions are identical. For example,
an inline function or an inline variable (since C++17) may be defined in a header file that is
#include'd in multiple source files.
```